### DISCUSSION TITLE
VM Exception while processing transaction: revert cannot stop indexing with an ongoing service agreement

### DISCUSSION THREAD

**Discussion by suntzu93** [subquery/subql#959] [Frontier TestNet - Bug Reporting]
I'm trying to Announce NotIndexing but I got error 

> VM Exception while processing transaction: revert cannot stop indexing with an ongoing service agreement

I don't have an ongoing service agreement.

<img width="1192" alt="Screen Shot 2022-05-03 at 16 05 53" src="https://user-images.githubusercontent.com/5988002/166428695-8dbb725b-7ae6-4208-b9c8-2753424b5518.png">

**Comment by suntzu93** [comment]
Seems to be the cause of my error when processing expired invoices.

> 2022-05-04T00:50:31.329Z <transaction> INFO remove expired service agreement: PROCESSING service agreement: 0x35C3a3ac40Cb7884c19b4b1bB5d8F30fbdaE6fcb 

> 2022-05-04T00:50:34.031Z <transaction> WARN remove expired service agreement: FAILED : Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"error\":{\"code\":-32603,\"message\":\"execution fatal: Module { index: 10, error: 0, message: None }\"},\"id\":12412}\n","error":{"code":-32603},"requestBody":"{\"method\":\"eth_estimateGas\",\"params\":[{\"type\":\"0x2\",\"maxFeePerGas\":\"0xd09dc300\",\"maxPriorityFeePerGas\":\"0x59682f00\",\"from\":\"0x653410c560e92bd0f113a78f681644637429239d\",\"to\":\"0xd098aace6495745496428085be2976391c6a5bc4\",\"data\":\"0x48ba6843000000000000000000000000de5a856a43b40173ea469f18bed962131ecebe0f0000000000000000000000000000000000000000000000000000000000000002\"}],\"id\":12412,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://sqtn.api.onfinality.io/public"}, tx={"data":"0x48ba6843000000000000000000000000de5a856a43b40173ea469f18bed962131ecebe0f0000000000000000000000000000000000000000000000000000000000000002","to":{},"from":"0x653410c560E92bd0F113a78f681644637429239D","type":2,"maxFeePerGas":{"type":"BigNumber","hex":"0xd09dc300"},"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"nonce":{},"gasLimit":{},"chainId":{}}, code=UNPREDICTABLE_GAS_LIMIT, version=abstract-signer/5.6.0)

**Comment by mzxyz** [comment]
The reason for the transaction failed error is due to not enough balance for your controller account, please get some DEV token in faucet and the error should be disappeared. Let me if you still have this issue after top up the account.

**Reply by suntzu93** [reply]
Thanks , resolved issue
